lib/phoenix/test/conn_test.ex

```diff
@@ -410,7 +410,7 @@ defmodule Phoenix.ConnTest do
       assert "can't be blank" in body["errors"]
 
   """
-  @spec json_response(Conn.t, status :: integer | atom) :: map | no_return
+  @spec json_response(Conn.t, status :: integer | atom) :: term | no_return
```

Comment:
while at it, consider dropping ` | no_return`. :) `no_return` is for functions that _never_ return: they always raise/throw/exit, stop the VM, loop infinitely, etc. If we were to add ` | no_return` to functions that may raise, we'd have to do that for majority of them as most would on invalid args, etc.